src/com/facebook/buck/apple/AppleBinaryDescription.java

```diff
@@ -396,13 +398,36 @@ private boolean shouldUseStubBinary(BuildRuleParams params) {
       BuildRuleResolver resolver,
       A args,
       Class<U> metadataClass) throws NoSuchBuildTargetException {
-    CxxBinaryDescription.Arg delegateArg = delegate.createUnpopulatedConstructorArg();
-    AppleDescriptions.populateCxxBinaryDescriptionArg(
-        new SourcePathResolver(resolver),
-        delegateArg,
-        args,
-        buildTarget);
-    return delegate.createMetadata(buildTarget, resolver, delegateArg, metadataClass);
+    if (!metadataClass.isAssignableFrom(FrameworkDependencies.class)) {
+      CxxBinaryDescription.Arg delegateArg = delegate.createUnpopulatedConstructorArg();
+      AppleDescriptions.populateCxxBinaryDescriptionArg(
+          new SourcePathResolver(resolver),
+          delegateArg,
+          args,
+          buildTarget);
+      return delegate.createMetadata(buildTarget, resolver, delegateArg, metadataClass);
+    }
+
+    Optional<Flavor> cxxPlatformFlavor = delegate.getCxxPlatforms().getFlavor(buildTarget);
+    Preconditions.checkState(
```

Comment:
If a bundle depends on another bundle (for instance, an app bundle with an app extension bundle as dependecy), this method is reached via `createAppleBundle() -> AppleBundle.createMetadata() -> AppleBinary.createMetadata()`, the platform flavor may not be set on this build target because a user doesn't usually annotate a bundle's binary parameter with a pre-existing platform.

This also doesn't take into account of multi-architecture binaries.